We start phase04 for pracrice unit test

hanaNrvtn opened the pull request.

Commit:
We start phase04 for pracrice unit test

AmirRezaZahedi committed:
debug BuildWordsBySign_ShouldGroupWordsBySign function

AmirRezaZahedi committed:
Improve unit tests and rename functions

- Enhanced unit tests for QueryFormatter and DocumentBuilder.
- Updated function names to follow consistent naming conventions for better clarity.

AmirRezaZahedi committed:
Debug InvertedIndexMap

tahadavari commented on phase04/FullTextSearch.Test/DocumentTest/DocumentBuilderTest.cs:
```diff
@@ -0,0 +1,82 @@
+        public DocumentBuilderTest()
+        {
+            _documentFormatter = Substitute.For<IDocumentFormatter>();
+            _documentBuilder = new DocumentBuilder(_documentFormatter);
```
_sut

AmirRezaZahedi replied:
solve

tahadavari commented on phase04/FullTextSearch.Test/DocumentTest/DocumentBuilderTest.cs:
```diff
@@ -0,0 +1,82 @@
+using Assert = Xunit.Assert;
+
+namespace FullTextSearch.Test.DocumentTest
+{
```
چرا when ندارن هیچ کدوم 
FunctionName_Should_When

AmirRezaZahedi replied:
solve

tahadavari commented on phase04/FullTextSearch.Test/MapTest/InvertedIndexMapperTest.cs:
```diff
@@ -0,0 +1,70 @@
+                { "mohammad", new List<Document> { document2, document3 } }
+            };
+
+            var mapper = new InvertedIndexMapper();
```
_sut مگه نساختید ؟

AmirRezaZahedi replied:
😲

tahadavari commented on phase04/FullTextSearch.Test/QueryTest/QueryFormatter.cs:
```diff
@@ -0,0 +1,59 @@
+        }
+        
+        [Xunit.Theory]
+        [InlineData(new[] { "+ali", "+mohammad" }, new[] { "+ali", "-reza", "zahra", "+mohammad" }, '+')]
```
اینو با MemberData فک کنم زیبا تر بشه نوشت

AmirRezaZahedi replied:
solve

tahadavari commented on phase04/FullTextSearch.Test/QueryTest/QueryFormatter.cs:
```diff
@@ -0,0 +1,59 @@
+        }
+
+        [Xunit.Theory]
+        [InlineData(new[] { "ali", "reza", "zahra", "mohammad" }, new[] { "+ali", "+reza", "+zahra", "+mohammad" })]
```
ایناهم چون خیلی طولانی و زیاد شدن شاید بد نباشه ClassData بزنید

AmirRezaZahedi replied:
what is diffrence beetwen **ClassData**  and **MemberData** ?

tahadavari commented on phase04/FullTextSearch.Test/QueryTest/QueryFormatter.cs:
```diff
@@ -0,0 +1,59 @@
+
+namespace InvertedIndex.Test.Query
+{
+    public class QueryFormatterTest
```
اسم فایل اصلاح بشه

tahadavari commented on phase04/FullTextSearch.Test/QueryTest/QueryFormatter.cs:
```diff
@@ -0,0 +1,59 @@
+        private readonly QueryFormatter _sut;
+
+        public QueryFormatterTest()
+        {
```
تستای این کلاس چرا AAA نیستند <؟

AmirRezaZahedi replied:
چون نیاز به arrange نیست

AmirRezaZahedi replied:
نیازه؟

AmirRezaZahedi committed:
Clean up debug code and add unit tests for SomeClass

- Removed debug code from previous commit.
- Added new unit tests for SomeClass to ensure functionality.

Commit:
done with unit test. Coverag: 84%

AmirRezaZahedi committed:
some bug solved and comments apply(InitializeServicetest.cs has problem now

AmirRezaZahedi committed:
change file's name

AliSK81 commented on phase04/FullTextSearch.Controller/DocumentController/Abstraction/IDocumentDirector.cs:
```diff
@@ -0,0 +1,6 @@
+
+public interface IDocumentDirector
+{
+    public void Construct(string name, string path, string text, IDocumentBuilder documentBuilder);
```
توی اینترفیس public نگذاریم
چون ذاتا هرچیزی داخل اینترفیس پابلیکه

AmirRezaZahedi replied:
solve

AliSK81 commented on phase04/FullTextSearch.Controller/DocumentController/Abstraction/IDocumentFormatter.cs:
```diff
@@ -0,0 +1,8 @@
+
+public interface IDocumentFormatter
+{
+    public string ToUpper(string text);
+    public IEnumerable<string> Split(string queryText, string regex);
+    public IEnumerable<string> RemoveStopWords();
```
remove public

AmirRezaZahedi replied:
solve

AliSK81 commented on phase04/FullTextSearch.Controller/DocumentController/DocumentBuilder.cs:
```diff
@@ -0,0 +1,42 @@
+    public DocumentBuilder(IDocumentFormatter documentFormatter)
+    {
+        _document = new Document();
+        _documentFormatter = documentFormatter;
```
check for argument null exception

AliSK81 commented on phase04/FullTextSearch.Controller/DocumentController/DocumentFormatter.cs:
```diff
@@ -0,0 +1,22 @@
+
+    public IEnumerable<string> RemoveStopWords()
+    {
+        throw new NotImplementedException();
```
اگه این تابع نیازی به پیاده سازی نداره، میتونیم بگیم طراحیمون خوب نبوده و شاید Interface segregationراه حل باشه

AmirRezaZahedi replied:
یه بار پاکش کرده بودیم اون اواسط فاز قبلی اصلا حواسم نبود بهش کامیتش کنم

AliSK81 commented on phase04/FullTextSearch.Controller/QueryController/QueryBuilder.cs:
```diff
@@ -0,0 +1,40 @@
+
+    public void BuildWordsBySign(IEnumerable<char> signs)
+    {
+        var splittedText = _queryFormatter.Split(_queryFormatter.ToUpper(_query.Text), " "); // [^A-Z]+
```
کامنت توی کد خوب نیست

AmirRezaZahedi replied:
100%

AliSK81 commented on phase04/FullTextSearch.Test/ControllerTest/SearchControllerTest/MinusSearcherTest.cs:
```diff
@@ -0,0 +1,64 @@
+            {' ', new List<string>(){"cat"}},
+        };
+
+        Document document1 = new Document();
+        document1.Name = "Doc1";
+        document1.Path = "./ResourcesTest/Doc1";
+        document1.Text = "reza ali mohammad hello";
+        document1.Words = new List<string> {"cat", "reza", "hello"};
```
var document1 = new Document
{
    Name = "Doc1",
    Path = "./ResourcesTest/Doc1",
    Text = "reza ali mohammad hello",
    Words = new List<string> { "cat", "reza", "hello" }
};

tahadavari replied:
عه علیم گذاشته بود

AmirRezaZahedi replied:
😲

AliSK81 commented on phase04/FullTextSearch.Test/ServiceTest/InitializeServiceTest/InitializeServiceTest.cs:
```diff
@@ -0,0 +1,102 @@
+
+        _directory.GetFiles(directoryPath).Returns(paths);
+
+        _path.GetFileName(paths[0]).Returns(names[0]);
+        _path.GetFileName(paths[1]).Returns(names[1]);
+        _path.GetFileName(paths[2]).Returns(names[2]);
+
+        _fileReader.ReadAsync(paths[0]).Returns(texts[0]);
+        _fileReader.ReadAsync(paths[1]).Returns(texts[1]);
+        _fileReader.ReadAsync(paths[2]).Returns(texts[2]);
+
+        _documentDirector.Construct(names[0], paths[0], texts[0], documentBuilders[0]);
+        _documentDirector.Construct(names[0], paths[0], texts[0], documentBuilders[1]);
+        _documentDirector.Construct(names[0], paths[0], texts[0], documentBuilders[2]);
+        
+        documents.Add(documentBuilders[0].GetDocument());
+        documents.Add(documentBuilders[1].GetDocument());
+        documents.Add(documentBuilders[2].GetDocument());
```
با for مینوشتیم اشکالی نداشت

AmirRezaZahedi replied:
solve

AliSK81 commented on phase04/FullTextSearch.Test/ServiceTest/SearchServiceTest/SearchServiceTest.cs:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    [Test]
+    public void Search_ShouldReturnResultForPassesQuery()
```
چی باعث میشه تست هامون بزرگ و long method بشن؟

AmirRezaZahedi replied:
کد های تکراری و وارد کردن دیتا های زیاد برای تست
شاید کاری که بشه با یک دیکشنری کرد داریم با چندتا میکنیم
بازم باید حضوری بحث کرد:)

AmirRezaZahedi replied:
ممکنه تابع ها چند کاره باشن ولی مال ما نبود تا حداکثرممکن

AliSK81 reviewed (approved):
خسته نباشید

tahadavari commented on phase04/FullTextSearch.Test/ControllerTest/ResultControllerTest/ResultBuilderTest.cs:
```diff
@@ -0,0 +1,123 @@
+        document2.Words = new List<string> {"cat", "reza", "demand"};
+            
+            
+        Document document3 = new Document();
```
var document3 = new Document
        {
            Name = "Doc3",
            Path = "./ResourcesTest/Doc3",
            Text = "reza ali mohammad hello",
            Words = new List<string> {"cat", "ali", "hello"}
        };

AmirRezaZahedi replied:
solve

tahadavari commented on phase04/FullTextSearch.Test/ControllerTest/ResultControllerTest/ResultBuilderTest.cs:
```diff
@@ -0,0 +1,123 @@
+        document2.Words = new List<string> {"cat", "reza", "demand"};
+            
+            
+        Document document3 = new Document();
```
var document3 = new Document
        {
            Name = "Doc3",
            Path = "./ResourcesTest/Doc3",
            Text = "reza ali mohammad hello",
            Words = new List<string> {"cat", "ali", "hello"}
        };

tahadavari replied:
و بقیه جاها

AmirRezaZahedi replied:
solve

AmirRezaZahedi committed:
some changing to make more clean code

AmirRezaZahedi committed:
Delete comments

AmirRezaZahedi committed:
:)
